### ISSUE TITLE
Streamline dependency management

### ISSUE THREAD

**Issue by NMertsch** [EuroPython/discord#138]
Currently, this project uses [pipenv](https://github.com/pypa/pipenv).

Maybe this is a personal skill issue, but at least half of the time I spent contributing to this project was spent fighting with pipenv. Other contributors also mentioned that pipenv is causing major trouble for them.

**Comment by egeakman** [comment]
Yep, I am one of those contributors. It makes the setup harder for me, and I manually install all the packages using my preferred env manager.

I think it would be better if we used plain pip-tools, or make this an installable package. Both of these let you install the deps to your preferred environment.

Big ➕ from me to change ``pipenv`` to something else.

**Comment by egeakman** [comment]
How does `uv` sound? :)

**Comment by NMertsch** [comment]
Let's collect requirements before we talk tools ;)
I'm not up to date with all the stuff going on in Python packaging land lately. I hear a lot of hype around uv, similar to pipenv and poetry before.
If pip, pip-freeze and requirements.txt solve all our problems, I'm fine with that as well.

This is what I think:

MUST:
- Easy to set up and use on Linux, Windows and MacOS, preferably without major impact of the system setup. Volunteers must be able to easily get started using their existing setup. They should not have to jump through many hoops to contribute.
- Stability. We can't predict the future, but it would be great if our 2025 setup still works in 2026 without major changes. Using standardized formats (based on PEPs) instead of tool-specific formats is the way to go.
- Dependency version locking for reproducibility.

SHOULD:
- Dependabot support or similar. We want packages to stay up to date without much manual intervention. If the tool causes more trouble than manual upgrades, we can do it manually.

Nice to have:
- Performance. Fast is nice, but slow is not a deal breaker. We don't install and update packages that often.

What do you think?

**Comment by NMertsch** [comment]
After looking into it deeper, I agree that `uv` seems to be a reasonable choice:
* Setup:
    * uv: Lots of options, including downloading a single `uv` binary from their GitHub releases
    * Python: uv can install and manage Python versions
    * Summary: Easy to use in all environments (all major OSs, no admin privileges required)
* Easy of use:
    * Good documentation
    * Simple set of commands
* Stability, Compability, Vendor Lock-In:
    * uv supports many build backends
    * uv uses standardized formats when possible:
        * Project metadata is managed in `pyproject.toml`'s `[project]` section ([PEP 621](https://peps.python.org/pep-0621/))
        * Package versions are still locked in `uv.lock`, but support for `pylock.toml` (https://peps.python.org/pep-0751/) is [in progress](https://github.com/astral-sh/uv/issues/12584)
    * uv is not required to install the package or run the code, so most contributions can be made using any Python installation with `venv` and `pip`

**Comment by egeakman** [comment]
We already use `uv` in [Program API](/EuroPython/programapi) and [Poirot](/EuroPython/internal-bot). Linked the repos in case it would be any help!

**Comment by NMertsch** [comment]
Thanks @egeakman, I just finished the PR and then saw your comment :D
